profile/feather_weight_benchmark.rb

```diff
@@ -0,0 +1,83 @@
+$LOAD_PATH.unshift(File.dirname(__FILE__))
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
+
+require 'mongo'
+require 'benchmark'
+require_relative 'benchmark_helper'
+
+##
+# Perform 'featherweight' benchmarks. This includes
+#
+# Common Flat BSON
+# Common Nested BSON
+# All BSON Types
+#
+# The featherweight benchmark is intended to test BSON serialization/deserialization
+#
+##
+def featherweight_benchmark!
+  #bench_helper = BenchmarkHelper.new('perftest','corpus')
+  bench_helper = BenchmarkHelper.new('foo','bar')
+  database = bench_helper.database
+  collection = bench_helper.collection
+
+
+  ##
+  # Common Flat BSON
+  #
+  # - Load FLAT_BSON dataset
+  #
+  # Measure: Encode each document to a BSON byte-string and decode the BSON byte-string back.
+  #
+  ##
+  flat_data = BenchmarkHelper.load_array_from_file('FLAT_BSON.txt')
+
+  first = Benchmark.bm do |bm|
+    bm.report('Featherweight::Common Flat BSON') do
+      flat_data.each do |doc|
+        BSON::Document.from_bson(  BSON::Document.new(doc).to_bson  )
+      end
+    end
+  end
```

Comment:
I would put each one of these chunks into a separate method and define the file names as constants at the top.

```ruby
FILES = ['FLAT_BSON.txt', 'DEEP_BSON.txt', 'FULL_BSON',txt]

def featherweight_benchmark!
  results = FILES.each do |file|
    data = BenchmarkHelper.load_array_from_file(file)

    Benchmark.bm do |bm|
      bm.report('..') do
       data.each do |doc|
         BSON::Document.from_bson(  BSON::Document.new(doc).to_bson  )
       end
     end
  end
end
```

not perfect, but hopefully you get what I'm saying.